homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -995,6 +1049,10 @@ def __init__(self, name, device, model, unique_id):
                 for mode in AirhumidifierOperationMode
                 if mode is not AirhumidifierOperationMode.Strong
             ]
+        elif self._model in [MODEL_AIRHUMIDIFIER_CA4]:
+            self._device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
+            self._available_attributes = AVAILABLE_ATTRIBUTES_AIRHUMIDIFIER_CA4
+            self._speed_list = [mode.name for mode in AirhumidifierMiotOperationMode]
```

Comment:
The speed list may only contain the official speed constants.

Reply:
same as above